frontend/src/scenes/insights/insightLogic.ts

```diff
@@ -43,6 +46,66 @@ export const insightLogic = kea<insightLogicType>({
                 setAllFilters: (_, { filters }) => filters,
             },
         ],
+        /*
+        isFirstLoad determines if this is the first graph being shown after the component is mounted (used for analytics)
+        */
+        isFirstLoad: [
+            true,
+            {
+                setNotFirstLoad: () => false,
+            },
+        ],
+    }),
+    listeners: ({ values, actions }) => ({
+        setAllFilters: (filters) => {
+            actions.reportUsage(filters.filters)
+            actions.setNotFirstLoad()
+        },
+        reportUsage: async (filters: Record<string, any>, breakpoint) => {
+            // Reports `insight viewed` event
+            const { insight, display, interval, date_from, date_to } = filters
+            const properties: Record<string, any> = {
```

Comment:
this is `Record<string, string | number` right? :)

Reply:
Will continue discussion on https://github.com/PostHog/posthog/pull/2787#discussion_r544891453